native/common/jp_booleantype.cpp

```diff
@@ -0,0 +1,199 @@
+/*****************************************************************************
+   Copyright 2004-2008 Steve Menard
+  
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+  
+*****************************************************************************/ 
+#include <jp_primitive_common.h>
+
+HostRef* JPBooleanType::asHostObject(jvalue val)
+{
+	if (val.z)
+	{
+		return JPEnv::getHost()->getTrue();
+	}
+	return JPEnv::getHost()->getFalse();
+}
+
+HostRef* JPBooleanType::asHostObjectFromObject(jvalue val)
+{
+	bool z = JPJni::booleanValue(val.l);
+	if (z)
+	{
+		return JPEnv::getHost()->getTrue();
+	}
+	return JPEnv::getHost()->getFalse();
+}
+
+EMatchType JPBooleanType::canConvertToJava(HostRef* obj)
+{
+	if (JPEnv::getHost()->isInt(obj) || JPEnv::getHost()->isLong(obj))
+	{
+		return _implicit;
+	}
+
+	if (JPEnv::getHost()->isWrapper(obj))
+	{
+		JPTypeName name = JPEnv::getHost()->getWrapperTypeName(obj);
+		if (name.getType() == JPTypeName::_boolean)
+		{
+			return _exact;
+		}
+	}
+
+	return _none;
+}
+
+jvalue JPBooleanType::convertToJava(HostRef* obj)
```

Comment:
There was a lot of drift in the convert to java methods.  We can clean them up and review them later but for now splitting it out makes it easy to see the issues.